document/CoreFormat.mdk

```diff
@@ -491,22 +533,57 @@ This allows tools to include information about the themselves that is not explic
 A `file` object represents a single file. In some cases, this file might be nested within another file,
 referred to as the "containing file."
 
+`file` objects occur as members of arrays within the `run.files` object (&sect;[#run-files]). 
+
 If the `file` object does not represent a nested file, neither of the properties
-`pathFromParent` (&sect;[#file-pathFromParent]) or `offsetFromParent` (&sect;[#file-offsetFromParent]) shall be present.
+`uri` (&sect;[#file-uri]) or `offsetFromParent` (&sect;[#file-offsetFromParent]) shall be present.
 
 If the location of a nested file can be expressed either by means of a path within the containing file
-or by means of a byte offset from the start of the containing file, then either `pathFromParent`
+or by means of a byte offset from the start of the containing file, then either `uri`
 or `offsetFromParent` (&sect;[#file-offsetFromParent]) or both shall be present; they must not both be absent.
 
-### `pathFromParent` property { #file-pathFromParent }
+### `uri` property { #file-uri }
 
 If the location of a nested file with respect to its containing file can be expressed
 only by means of a path within its containing file,
```

Comment:
We could certainly have done that, but we did examine this in detail when we de-dup'd the physical location information, and this is what we decided. We can certainly revisit that decision. It's a balance between on the one hand, making life easier for producers by not requiring them to compute and emit information that the consumer doesn't strictly need, and on the other hand, making life easier for consumers by giving them as much information as possible, even if they don't need all of it.

In this particular case... I don't know. It seems that it would be a rare case where the producer knew both the offset and the uri. Your call.